lemminx-maven/src/main/java/org/eclipse/lemminx/maven/SubModuleValidator.java

```diff
@@ -32,23 +33,19 @@ public void setPomFile(File pomFile) throws FileNotFoundException, IOException,
 		model = mavenreader.read(new FileReader(pomFile));
 	}
 
-	public Diagnostic validateSubModuleExistence(DiagnosticRequest diagnosticRequest) {
+	public Optional<Object> validateSubModuleExistence(DiagnosticRequest diagnosticRequest) {
 		DOMNode node = diagnosticRequest.getNode();
-		DOMDocument xmlDocument = diagnosticRequest.getXMLDocument();
-		Diagnostic diagnostic = null;
-		Range range = diagnosticRequest.getRange();
 		String tagContent = null;
 		if (node.hasChildNodes()) {
 			tagContent = node.getChild(0).getNodeValue(); // tagContent is the module to validate eg.
 															// <module>tagContent</module>
 		}
 		if (node.hasChildNodes() && !model.getModules().contains(tagContent)) {
-			diagnostic = new Diagnostic(range, String.format("Module '%s' does not exist", tagContent),
-					DiagnosticSeverity.Error, xmlDocument.getDocumentURI(), "XML");
-			diagnosticRequest.getDiagnostics().add(diagnostic);
+			List<Diagnostic> diagnostics = new ArrayList<>();
```

Comment:
You can use `Collections.singletonList(diagnosticRequest.createDiagnostic(....))` instead of manually instantiating a new ArrayList. Singleton collections will consume slightly less memory.

Reply:
Thanks I've wondered what was the nicer way of doing that. Will adjust.